DROD/RoomWidget.cpp

```diff
@@ -1039,8 +1039,16 @@ void CRoomWidget::HighlightSelectedTile()
 				//Show citizen's current destination.
 				const CCitizen *pCitizen = DYN_CAST(const CCitizen*, const CMonster*, pMonster);
 				UINT wDestX, wDestY;
-				if (pCitizen->GetGoal(wDestX,wDestY))
+				if (pCitizen->GetGoal(wDestX, wDestY)) {
 					AddShadeEffect(wDestX, wDestY, PaleYellow);
+					for (UINT wPathIndex = 0; wPathIndex < pCitizen->GetStoredPath().GetSize(); ++wPathIndex)
```

Comment:
I didn't mention this before on the Halph path PR, but the code calling `GetStoredPath()` would be more efficient and cleaner if the result were saved in a local variable, so the data structure isn't re-copied from the monster object twice in each iteration of these loops (an N-squared cost).